### DISCUSSION TITLE
multiple backupstoragelocation configs

### DISCUSSION THREAD

**Discussion by badalk** [vmware-tanzu/velero#3714] [Community support Q&A]
Looks like with velero:v1.6.0 it is possible to provide multiple backupstoragelocation. I want to use this feature but looking for some guidance around how to do it through configuration rather than through velero CLI? 
Context:

- I have a multi-tenant cluster where each application is a tenant. They may have different storage accounts and associated identities (in my case I am using pod identity). Good news is v1.6.0 supports this
- I am also using FluxCD with helmoperator for GitOps, and hence deploying velero chart through helm-operator
- Looks like velero deployment needs to have at least one backupstoragelocation and cannot be deployed without any

What i was looking to do is this:

- To deploy velero chart, without any backupstoragelocation at first
- As add new tenants into the cluster, create new backupstoragelocation for their tenant, and create schedule for them based on their need and patch it on top of the default velero deployment
- And i want to do all this through configurations.
- The velero-plugin-for-microsoft-azure seems to have a resource config based on BackupStorageLocation CRD but how do i create this and apply in the velero deployment? and what if i have it separate for each tenant? do we have any exmaple on how to apply those yamls for BackupStorageLocation and Schedules after velero deployement?

**Comment by jenting** [comment]
I saw you create an [issue](https://github.com/vmware-tanzu/helm-charts/issues/254) in helm-chart repo already.
We had put it in our backlog and will try to enhance the helm chart to support configures multiple  BSL credentials during helm install.

**Reply by badalk** [reply]
any idea by when this will be available?

**Reply by jenting** [reply]
I need to check if the helm chart supports multiple BSL configurations, is it comply backward-compatible?
